src/services/codefixes/requireInTs.ts

```diff
@@ -66,25 +70,27 @@ interface Info {
     readonly defaultImportName: Identifier | undefined;
     readonly namedImports: NamedImports | undefined;
     readonly statement: VariableStatement;
-    readonly required: StringLiteralLike;
+    readonly moduleSpecifier: StringLiteralLike;
 }
 
-function getInfo(sourceFile: SourceFile, program: Program, pos: number): Info | undefined {
+function getInfo(sourceFile: SourceFile, program: Program, pos: number, preferences: UserPreferences): Info | undefined {
     const { parent } = getTokenAtPosition(sourceFile, pos);
     if (!isRequireCall(parent, /*requireStringLiteralLikeArgument*/ true)) {
         Debug.failBadSyntaxKind(parent);
     }
 
     const decl = cast(parent.parent, isVariableDeclaration);
+    const quotePreference = getQuotePreference(sourceFile, preferences);
     const defaultImportName = tryCast(decl.name, isIdentifier);
     const namedImports = isObjectBindingPattern(decl.name) ? tryCreateNamedImportsFromObjectBindingPattern(decl.name) : undefined;
     if (defaultImportName || namedImports) {
+        const moduleSpecifier = first(parent.arguments);
         return {
             allowSyntheticDefaults: getAllowSyntheticDefaultImports(program.getCompilerOptions()),
             defaultImportName,
             namedImports,
             statement: cast(decl.parent.parent, isVariableStatement),
-            required: first(parent.arguments),
+            moduleSpecifier: isNoSubstitutionTemplateLiteral(moduleSpecifier) ? factory.createStringLiteral(moduleSpecifier.text, quotePreference === QuotePreference.Single) : moduleSpecifier,

```

Comment:
Maybe this isn't a new bug, but we reject dynamic requires like `require(`node:${a}`)`, yes?

Reply:
There is test that covers this case  https://github.com/microsoft/TypeScript/blob/ce1c83ebc38b2ebf7a0a9806f78ee5e6bfab69d9/tests/cases/fourslash/codeFixRequireInTs5.ts